shared/util.go

```diff
@@ -634,6 +634,15 @@ func StringInSlice(key string, list []string) bool {
 	return false
 }
 
+func StringPrefixInSlice(key string, list []string) bool {
+	for _, entry := range list {
+		if strings.HasPrefix(entry, key) {
+			return true
+		}
+	}
+	return false
```

Comment:
Please add a newline before `return false`, as functional blocks should be followed by a newline. Our `newline-after-block` linter unfortunately doesn't catch this for `for` loops.